### PR TITLE
feat: rebasing dshares

### PR DESCRIPTION
**Pull request by jaketimothy**
Deep refactor for handling splits based on rebasing dShare tokens wrapped in xdShare vaults for compat.

Includes update of OpenZeppellin to v5 and solidity to 0.8.22.

### CODE REVIEW

**Comment by jaketimothy** [comment]
Task linked: [CU-86868xcu5 Rebasing dshares](https://app.clickup.com/t/86868xcu5)

File: test/main/xdShare.t.sol [hunk 1]
```diff
@@ -183,6 +183,68 @@ contract xdShareTest is Test {
         assertEq(token.balanceOf(address(xToken)), 0);
     }
 
+    // function testDepositRedeemSplit(uint128 supply, uint8 multiple, bool reverse) public {
```
Comment (joshualyguessennd) [inline test/main/xdShare.t.sol:186]: is it something we need to fix in another PR ?

**Review by joshualyguessennd — COMMENTED** [review]
it looks good

**Review by joshualyguessennd — APPROVED** [review]

File: test/main/xdShare.t.sol (hunk 1 above)
Comment (jaketimothy) [reply test/main/xdShare.t.sol]: fixed

**Pull request merged**